sprocs/ip_to_mode.sql

```diff
@@ -15,5 +18,17 @@ BEGIN
     ELSE
         mode := 'Public';
     END IF;
+
+    -- Do we have an active CBTF reservation, if so force mode to 'Exam'
+    PERFORM *
+    FROM reservations AS r
+    WHERE
+        r.user_id = ip_to_mode.authn_user_id
+        AND r.delete_date IS NULL
```

Comment:
Do we also want `AND r.checked_in IS NOT NULL` here?

Reply:
I agree - without something like that, wouldn't it be the case that users would be forced into exam mode for the duration of the reservation, regardless of whether they were actually trying to take the exam? This seems to be a problem in particular for remote users - but I don't know enough about how the CBTF is handling remote users these days. (Do they check in and out?)

Reply:
It's a bit of a sticky point that there's no check out. There's only `checked_in` and `access_start` and `access_end`  (all are timestamps). A checked out session would be one where `checked_in IS NOT NULL and now() > access_end` - but we don't have an integrated way for students to "checkout", it just times out.

That didn't matter so much when the checkin was restricted to CBTF IP space. (If their reservation went longer and they left the CBTF, they were still mode:Public from other IP addresses.) Here I'm thinking we might want to at least put a notice (maybe in pages/home?) and potentially give them a button to end exam mode to get normal mode:Public back. Thoughts on that?